src/controllers/portfolio/portfolio.test.ts

```diff
@@ -88,48 +88,150 @@ describe('Portfolio Controller ', () => {
-      await controller.updateSelectedAccount([account2], account2.addr)
-      const storagePreviousHints = await storage.get('previousHints', {})
-      const ethereumHints = storagePreviousHints.fromExternalAPI[`ethereum:${account2.addr}`]
-      const polygonHints = storagePreviousHints.fromExternalAPI[`polygon:${account2.addr}`]
-      const optimismHints = storagePreviousHints.fromExternalAPI[`polygon:${account2.addr}`]
-
-      // Controller persists tokens having balance for the current account.
-      // @TODO - here we can enhance the test to cover one more scenarios:
-      //  #1) Does the account really have amount for the persisted tokens.
-      expect(ethereumHints.erc20s.length).toBeGreaterThan(0)
-      expect(Object.keys(ethereumHints.erc721s).length).toBeGreaterThan(0)
-      expect(polygonHints.erc20s.length).toBeGreaterThan(0)
-      expect(optimismHints.erc20s.length).toBeGreaterThan(0)
+      .mockImplementationOnce(
+        () =>
+          new Promise((resolve) => {
+            setTimeout(() => {
+              queueOrder.push('updatePortfolioState - #1 call (pending state)')
+              resolve(true)
+            }, 2000)
+          })
+      )
+      .mockImplementationOnce(
+        () =>
+          new Promise((resolve) => {
+            queueOrder.push('updatePortfolioState - #2 call (latest state)')
+            resolve(true)
+          })
+      )
+      .mockImplementationOnce(
+        () =>
+          new Promise((resolve) => {
+            queueOrder.push('updatePortfolioState - #2 call (pending state)')
+            resolve(true)
+          })
+      )
+      .mockImplementationOnce(
+        () =>
+          new Promise((resolve) => {
+            queueOrder.push('updatePortfolioState - #3 call (latest state)')
+            resolve(true)
+          })
+      )
+      .mockImplementationOnce(
+        () =>
+          new Promise((resolve) => {
+            queueOrder.push('updatePortfolioState - #3 call (pending state)')
+            resolve(true)
+          })
+      )
+
+    controller.updateSelectedAccount([account], account.addr, ethereum, undefined, {
+      forceUpdate: true
+    })
+
+    controller.updateSelectedAccount([account], account.addr, ethereum, undefined, {
+      forceUpdate: true
+    })
+
+    // We need to wait for the latest update, or the bellow expect will run too soon,
+    // and we won't be able to check the queue properly.
+    await controller.updateSelectedAccount([account], account.addr, ethereum, undefined, {
+      forceUpdate: true
     })
+
+    expect(queueOrder).toEqual([
+      'updatePortfolioState - #1 call (latest state)',
+      'updatePortfolioState - #1 call (pending state)',
+      'updatePortfolioState - #2 call (latest state)',
+      'updatePortfolioState - #2 call (pending state)',
+      'updatePortfolioState - #3 call (latest state)',
+      'updatePortfolioState - #3 call (pending state)'
+    ])
```

Comment:
Can't this be simplified to something along the line of:
![image](https://github.com/AmbireTech/ambire-common/assets/68795596/8b030507-1277-4e92-8dfc-d6c4789fa52e)

Reply:
unfortunately no, because of 2 reasons:

1. we can't delay one of the calls, as we did it with the mock
2. updateSelectedAccount resolves when all networks are resolved, but the queue is implemented per network and account, and we need to test a single queue.